Packs/FeedUnit42v2/Integrations/FeedUnit42v2/FeedUnit42v2.py

```diff
@@ -161,29 +180,127 @@ def parse_malware(client, malware_objects: list = [], feed_tags: list = [],
     return malware_list
 
 
-def get_campaign_from_sub_reports(report_object, id_to_object):
-    report_relationships = []
-    object_refs = report_object.get('object_refs', [])
-    for obj in object_refs:
-        if obj.startswith('report--'):
-            sub_report_obj = id_to_object.get(obj, {})
-            for sub_report_obj_ref in sub_report_obj.get('object_refs', []):
-                if sub_report_obj_ref.startswith('campaign--'):
-                    related_campaign = id_to_object.get(sub_report_obj_ref)
-
-                    if related_campaign:
-                        entity_relation = EntityRelationship(name='related-to',
-                                                             entity_a=f"[Unit42 ATOM] {report_object.get('name')}",
-                                                             entity_a_type='Report',
-                                                             entity_b=related_campaign.get('name'),
-                                                             entity_b_type='Campaign')
-                        report_relationships.append(entity_relation.to_indicator())
-    return report_relationships
-
-
-def is_sub_report(report_obj):
+def is_atom42_main_report(report_obj):
+    """ Our definition for main report is a report with 'object_refs'
+    list that contains only objects of the type not a report and a report with
+    description.
+
+    Args:
+      report_obj: The report object.
+    Returns:
+        A boolean.
+    """
+    obj_refs = report_obj.get('object_refs', [])
+    contain_report = False
+    contain_intrusion_set = False
+    if report_obj.get('name') == 'ATOM Campaign Report':
+        return False
+    for obj_ref in obj_refs:
+        if not obj_ref.startswith(('report--', 'intrusion-set--')):
+            return False
+        if obj_ref.startswith('intrusion-set--'):
+            contain_intrusion_set = True
+        if obj_ref.startswith('report--'):
+            contain_report = True
+    return contain_intrusion_set and contain_report
+
+
+def is_atom42_sub_report(report_obj):
+    """ Our definition for sub report is a report with 'object_refs'
+    list that contains only objects not for type intrusion-set or report.
+
+    Args:
+      report_obj: The report object.
+    Returns:
+        A boolean.
+    """
     obj_refs = report_obj.get('object_refs', [])
-    return all(not obj_ref.startswith('report--') for obj_ref in obj_refs)
+    is_report_description = report_obj.get('description') is not None
+    only_reports_and_intrusion_set = True
+    if report_obj.get('name') == 'ATOM Campaign Report':
+        return True
+    for obj_ref in obj_refs:
+        if not obj_ref.startswith(('report--', 'intrusion-set--')):
+            only_reports_and_intrusion_set &= False
```

Comment:
Why not set it to Flase and that's it? there is not a chance it will be False before this line as we are setting it to True in 219

Reply:
Correct, Changed.